tutorials/epochs/baseline_regression.py

```diff
@@ -0,0 +1,185 @@
+
+del raw_filtered  # free up memory
+
+# %% we average the epochs per condition and baseline correct the data
+
+auditory_baseline = epochs[epochs.events[:, 2] == 1].copy().\
+    apply_baseline((baseline_tmin, baseline_tmax)).average()
+
+visual_baseline = epochs[epochs.events[:, 2] == 2].copy().\
+    apply_baseline((baseline_tmin, baseline_tmax)).average()
+
+# %% traditional baseline correction
+
+fig = mne.viz.plot_compare_evokeds({"auditory": auditory_baseline, "visual": visual_baseline},
+                                   picks=[ch],
+                                   invert_y=True,
+                                   title="traditional baseline correction",
+                                   show_sensors=False, legend=True,
+                                   colors=dict(auditory="k", visual="r"),
+                                   truncate_yaxis=False)
+
+# let's look at the topography of the contrast auditory - visual evoked response
+
+diff = mne.combine_evoked([auditory_baseline, visual_baseline], [1, -1])
+
+diff.plot_topomap(times=[0., 0.05, 0.1, 0.2, 0.3, 0.4], ch_type='eeg')
+
+# maybe check a different channel?
+
+diff.plot_topo()
+
+# %% regression based baseline correction
+
+baseline_epochs = epochs.copy().pick_channels([ch]).crop(baseline_tmin, baseline_tmax)
+epoch_data = baseline_epochs.get_data()
+
+baseline = epoch_data.mean(axis=-1).squeeze() * 1e6
+
+# we convert to millivolt to make is easier for the regression to fit the data
+
+# set up design matrix
+
+design_matrix = np.stack([epochs.events[:, 2] == 1, epochs.events[:, 2] == 2,
+                          baseline, baseline*(epochs.events[:, 2] == 2)]).T
+
+# shape should be number of epochs per subject x number of regressor (so in that example 286 x 4)
+
+# fit regression model
+
+# epochs should have 286 epochs
+
+# we fit the model for all channels
+
+regmodel = mne.stats.linear_regression(epochs, design_matrix,
+                                               names=["auditory", "visual",
+                                                      "baseline",
+                                                      "baseline:visual"])
+
+regbaseline_aud = regmodel['auditory'].beta
+regbaseline_vis = regmodel['visual'].beta
+
+# %% plot regression output
+
+fig = mne.viz.plot_compare_evokeds({'auditory': regbaseline_aud, 'visual': regbaseline_vis},
+                                   invert_y=True,
+                                   picks=ch,
+                                   title="baseline correction based on regression",
+                                   show_sensors=False,
+                                   colors=dict(auditory="k", visual="r"),
+                                   truncate_yaxis=False)
+
+# let's look at the topography of the contrast auditory - visual evoked response
+
+regressed_diff = mne.combine_evoked([regbaseline_aud, regbaseline_vis], [1, -1])
+
+regressed_diff.plot_topomap(times=[0., 0.05, 0.1, 0.2, 0.3, 0.4], ch_type='eeg')
```

Comment:
I found it very useful to plot `regmodel['baseline'].beta` as well. Than you can see how much baseline is removed for each time point.